### DISCUSSION TITLE
Is there a way to traverse a zip file's content like a JS Object?

### DISCUSSION THREAD

**Discussion by julillermo** [gildas-lormeau/zip.js#531] [Q&A]
Hello! 😁

Is there a way to navigate through the zip file's contents as if it were a JS Object?

I previously worked with the [BeautifulSoup Python library](https://beautiful-soup-4.readthedocs.io/en/latest/) for parsing and traversing HTML and HTML-like files. In BS4, you can go up and down the content hierarchy like a tree structure by accessing them with syntax very similar to how JavaScript handles objects.

For example, with an HTML structure like the following:

```html
...
<body>
     <div>
          <a><p>content</p></a>
     </div>
</body>
...
```
The `<p>` tag can be accessed with something like `body.div.a.p`

I wanted to achieve a similar functionality in my project, and I was curious to know if a feature or solution like this already exists.

Additional context:
*I'm working on a personal project where I handle zip files with an expected structure (potentially different file names and the contents inside the zip files themselves, but generally the same structure). I want to navigate to a file/directory and directly access and work with the contents from there.*

**Comment by gildas-lormeau** [comment]
There is the [`FS`](https://gildas-lormeau.github.io/zip.js/api/classes/FS.html) API for this, see for example https://github.com/gildas-lormeau/zip.js/blob/4137ff5bff894568a369de2bea6291ca0e2eab40/tests/all/test-fs.js.
